dev-haskell/cpphs/cpphs-1.20.9.1.ebuild

```diff
@@ -24,3 +24,11 @@ RDEPEND=">=dev-haskell/old-locale-1.0.0.2:=[profile?] <dev-haskell/old-locale-1.
 DEPEND="${RDEPEND}
 	>=dev-haskell/cabal-1.8
 "
+
+src_prepare() {
+	default
+
+	cabal_chdeps \
+		'time >=1.5 && <1.11' 'time >=1.5'
+}
```

Comment:
Why not use the [upstream Hackage revision](https://hackage.haskell.org/package/cpphs-1.20.9.1/revisions/) instead?

`hackport` will add this automatically for you if you re-run

```bash
hackport merge cpphs
```

Reply:
Also, I want to point out that we have `CABAL_CHDEPS` now, so you don't need to make a custom `src_prepare`.

https://github.com/gentoo-haskell/gentoo-haskell/blob/dcc87ee4c96ac0a3c55985532953663cc1db0e05/eclass/haskell-cabal.eclass#L171-L181

Reply:
`src_prepare` is not as simple as it used to be and you might leave out some needed functionality if you override it in the ebuild without knowing what you're doing.

https://github.com/gentoo-haskell/gentoo-haskell/blob/dcc87ee4c96ac0a3c55985532953663cc1db0e05/eclass/haskell-cabal.eclass#L641-L662

Reply:
> Why not use the [upstream Hackage revision](https://hackage.haskell.org/package/cpphs-1.20.9.1/revisions/) instead?
> 
> `hackport` will add this automatically for you if you re-run
> 
> ```shell
> hackport update && hackport merge cpphs
> ```

can't use hackport with newer ghc, broken